modules/trusted-firmware-m/tfm_boards/board/RTE_Device.h

```diff
@@ -8,45 +8,37 @@
 #define __RTE_DEVICE_H
 
 #include <autoconf.h>
+
 /* ARRAY_SIZE causes a conflict as it is defined both by TF-M and indirectly by devicetree.h */
 #undef ARRAY_SIZE
 #include <zephyr/devicetree.h>
+
 #include <nrf-pinctrl.h>
 
-#define UART_PIN_INIT(node_id, prop, idx) \
-	DT_PROP_BY_IDX(node_id, prop, idx),
+/* Configuration settings for Driver_FLASH.c and Driver_USART.c */
 
-/* Configuration settings for Driver_USART0. */
-#if defined(CONFIG_TFM_SECURE_UART0) || DOMAIN_NS == 1U
+#define RTE_FLASH0 1
 
+#if defined(CONFIG_TFM_SECURE_UART0) || DOMAIN_NS == 1U
 #define RTE_USART0 1
+#endif
 
-#define RTE_USART0_PINS \
-{ \
-	DT_FOREACH_CHILD_VARGS( \
-		DT_PINCTRL_BY_NAME(DT_NODELABEL(uart0), default, 0), \
-		DT_FOREACH_PROP_ELEM, psels, UART_PIN_INIT \
-	) \
-}
-
-#endif /* defined(CONFIG_TFM_SECURE_UART0) || DOMAIN_NS == 1U */
-
-/* Configuration settings for Driver_USART1. */
-#if defined(CONFIG_TFM_SECURE_UART1) && DOMAIN_NS != 1U
-
+#if defined(CONFIG_TFM_SECURE_UART1)
 #define RTE_USART1 1
+#endif
 
-#define RTE_USART1_PINS \
+#define UART_PIN_INIT(node_id, prop, idx) \
+	DT_PROP_BY_IDX(node_id, prop, idx),
+
+#define RTE_USART_PINS(idx)						\
 { \
 	DT_FOREACH_CHILD_VARGS( \
-		DT_PINCTRL_BY_NAME(DT_NODELABEL(uart1), default, 0), \
+		DT_PINCTRL_BY_NAME(DT_NODELABEL(uart##idx), default, 0), \
 		DT_FOREACH_PROP_ELEM, psels, UART_PIN_INIT \
 	) \
 }
 
-#endif /* defined(CONFIG_TFM_SECURE_UART1) */
-
-/* Configuration settings for Driver_FLASH0. */
-#define RTE_FLASH0 1
+#define RTE_USART0_PINS RTE_USART_PINS(0)
```

Comment:
Is that correct? Before the create of the nodes were ifdefed by the TFM_SECURE_UART0/1 , right? 

Now you generate both of the nodes which also same the same names and idx:
DT_PINCTRL_BY_NAME(node_id, name, idx)

Does that make sense to define these inside the ifdefs?

Reply:
discussed offline